lib/ae_mdw/db/mutations/name_claim_mutation.ex

```diff
@@ -90,113 +90,107 @@ defmodule AeMdw.Db.NameClaimMutation do
-                Model.AuctionExpiration,
-                {prev_auction_end, plain_name}
+    timeout = :aec_governance.name_claim_bid_timeout(plain_name, protocol_version)
+
+    if timeout == 0 do
+      expire = Names.expire_after(height)
+
+      m_name =
+        Model.name(
+          index: plain_name,
+          active: height,
+          expire: expire,
+          owner: owner_pk,
+          auction_timeout: 0
+        )
+
+      lock_amount = (lima_or_higher? && name_fee) || :aec_governance.name_claim_locked_fee()
+
+      name_claim = Model.name_claim(index: {plain_name, height, txi_idx})
+
+      ObjectKeys.put_active_name(state, plain_name)
+
+      state2
+      |> Name.put_active(m_name)
+      |> State.put(Model.NameClaim, name_claim)
+      |> Name.delete_inactive(plain_name)
+      |> IntTransfer.fee({height, txi_idx}, :lock_name, owner_pk, txi_idx, lock_amount)
+      |> State.inc_stat(:burned_in_auctions, lock_amount)
+      |> Names.increment_names_count(owner_pk)
+    else
+      state3 =
+        IntTransfer.fee(state2, {height, txi_idx}, :spend_name, owner_pk, txi_idx, name_fee)
+
+      state4 =
+        case State.get(state3, Model.AuctionBid, plain_name) do
+          :not_found ->
+            auction_claim = Model.auction_bid_claim(index: {plain_name, height, txi_idx})
+            auction_end = height + timeout
+
+            m_auction_bid =
+              Model.auction_bid(
+                index: plain_name,
+                start_height: height,
+                block_index_txi_idx: {block_index, txi_idx},
+                expire_height: auction_end,
+                owner: owner_pk
               )
-              |> IntTransfer.fee(
-                {height, txi_idx},
-                :refund_name,
-                prev_owner,
-                prev_txi_idx,
-                prev_name_fee
-              )
-              |> State.inc_stat(:locked_in_auctions, name_fee - prev_name_fee)
-              |> State.put(Model.AuctionBidClaim, auction_claim)
-              |> State.put(Model.AuctionBid, m_auction_bid)
-              |> State.put(
-                Model.AuctionExpiration,
-                Model.expiration(index: {actual_auction_end, plain_name})
+
+            m_auction_exp = Model.expiration(index: {auction_end, plain_name})
+
+            state3
+            |> State.inc_stat(:auctions_started)
+            |> State.put(Model.AuctionBidClaim, auction_claim)
+            |> State.put(Model.AuctionBid, m_auction_bid)
+            |> State.put(Model.AuctionExpiration, m_auction_exp)
+
+          {:ok,
+           Model.auction_bid(
+             start_height: start_height,
+             block_index_txi_idx: {_bi, prev_txi_idx},
+             expire_height: prev_auction_end,
+             owner: prev_owner
+           ) = auction_bid} ->
+            auction_end =
+              prev_auction_end +
+                :aec_governance.name_claim_bid_extension(plain_name, protocol_version)
```

Comment:
The auction should only be extended if the bid is near the end (https://forum.aeternity.com/t/the-proposed-changes-in-ceres-protocol/12056#aens-adjusted-auction-parameters-7)

Reply:
It's extended when auction already exists, which is this case, am I missing anything? https://github.com/aeternity/aeternity/blob/master/apps/aeprimop/src/aeprimop.erl#L936

Reply:
It's ok in that case